### PR TITLE
add deprecation note.

### PR DESCRIPTION
**Pull request by tacman**
Having both amazon-s3 and aws-s3 is confusing.

### CODE REVIEW

File: Resources/doc/cache-resolver/amazons3.rst [hunk 1]
```diff
@@ -7,6 +7,8 @@ Amazon S3 Resolver
 The ``AmazonS3Resolver`` resolver enables cache resolution using the
 ``\AmazonS3`` storage API.
 
+This will be deprecated in Version 3, use aws-s3 resolver instead.
```
Comment (dbu) [inline Resources/doc/cache-resolver/amazons3.rst]: we deprecate it now and remove it in version 3, lets say:

```suggestion
**Deprecated**, use the aws-s3 resolver instead. The legacy AmazonS3Resolver will be removed in version 3.
```

**Comment by dbu** [comment]
do we need an S3 resolver at all in version 3? or could we do all with flysystem? or can flysystem not provide the web URL to a file on S3?

**Comment by tacman** [comment]
Hmm, interesting question.  Regardless, we need more documentation and examples.

This is what I use for s3 on production and local filesystem in dev.

```yaml
# config/packages/flysystem.yaml
# Read the documentation at https://github.com/thephpleague/flysystem-bundle/blob/master/docs/1-getting-started.md
flysystem:
    storages:
        default.storage:
            adapter: 'aws'
            # visibility: public # Make the uploaded file publicly accessible in S3
            options:
                client: 'Aws\S3\S3Client' # The service ID of the Aws\S3\S3Client instance
                bucket: '%env(AWS_S3_BUCKET_NAME)%'
                streamReads: true
                prefix: '%env(S3_STORAGE_PREFIX)%'
when@dev:
    flysystem:
        storages:
            default.storage:
                adapter: 'local'
                options:
                    directory: '%kernel.project_dir%/public/images'

```

And

```yaml
# config/packages/liip_imagine.yamn
liip_imagine:
    driver: "gd"
    twig:
        mode: lazy

    loaders:
        flysystem_loader:
            flysystem:
                # this comes from flysystem.yaml
                filesystem_service: default.storage

    # default loader to use for all filter sets
    data_loader: flysystem_loader
```

**Comment by dbu** [comment]
looks like modern versions of flysystem actually support that: https://flysystem.thephpleague.com/docs/usage/public-urls/

i think the way to go for 3.x is to leverage flysystem as much as we can to avoid this bundle having to care about storage and filesystem and such stuff.

**Pull request merged**